src/main/java/org/sqlite/jdbc4/JDBC4ResultSet.java

```diff
@@ -320,6 +325,55 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
         if (type == Time.class) return type.cast(getTime(columnIndex));
         if (type == Timestamp.class) return type.cast(getTimestamp(columnIndex));
 
+        if (type == LocalDate.class) {
+            try {
+                Date date = getDate(columnIndex);
+
+                int year = date.getYear() + 1900;
+                int month = date.getMonth() + 1;
+                int day = date.getDate();
+
+                return type.cast(LocalDate.of(year, month, day));
+            } catch (SQLException sqlException) {
+                // If the FastDateParser failed, try parse it with LocalDate.
+                // It's a workaround for a value like '2022-12-1' (i.e no time presents).
+
+                return type.cast(LocalDate.parse(getString(columnIndex)));
+            }
+        }
+
+        if (type == LocalTime.class) {
+            try {
+                Time time = getTime(columnIndex);
+
+                int hour = time.getHours();
+                int minute = time.getMinutes();
+                int second = time.getSeconds();
+
+                return type.cast(LocalTime.of(hour, minute, second));
+            } catch (SQLException sqlException) {
+                // If the FastDateParser failed, try parse it with LocalTime.
+                // It's a workaround for a value like '11:22:22' (i.e no date presents).
+
+                return type.cast(LocalTime.parse(getString(columnIndex)));
+            }
+        }
```

Comment:
same here, there is a [toLocalTime](https://docs.oracle.com/javase/8/docs/api/java/sql/Time.html#toLocalTime--) function